deployer/hub.py

```diff
@@ -41,6 +42,40 @@ def auth(self):
         else:
             raise ValueError(f'Provider {self.spec["provider"]} not supported')
 
+    def ensure_docker_credhelpers(self):
+        """
+        Setup credHelper for current hub's image registry.
+
+        Most image registries (like ECR, GCP Artifact registry, etc) use
+        a docker credHelper (https://docs.docker.com/engine/reference/commandline/login/#credential-helpers)
+        to authenticate, rather than a username & password. This requires an
+        entry per registry in ~/.docker/config.json.
+
+        This method ensures the appropriate credential helper is present
+        """
+        image_name = self.spec['image_repo']
+        registry = image_name.split('/')[0]
+
+        helper = None
+        # pkg.dev is used by Google Cloud Artifact registry
+        if registry.endswith('pkg.dev'):
+            helper = 'gcloud'
+
+        if helper is not None:
+            dockercfg_path = os.path.expanduser('~/.docker/config.json')
```

Comment:
I usually do not like to "mess" with user's config files but given your other comment about splitting this into their own repo, I think it is OK for now :wink: